blocks/monitoring/evaluators.py

```diff
@@ -5,13 +5,80 @@
 from theano import tensor
 
 from blocks.utils import dict_subset
-from blocks.monitoring.aggregation import _DataIndependent, Mean, TakeLast
+from blocks.monitoring.aggregation import (_DataIndependent, Mean,
+                                           TakeLast, MonitoredQuantity)
 from blocks.graph import ComputationGraph
 from blocks.utils import reraise_as
 
 logger = logging.getLogger()
 
 
+class MonitoredQuantityBuffer(object):
+    """Intermediate results of aggregating values of monitored-quantity.
+
+    Accumulate results for a list of monitored-quantity for every
+    single batch. Provides initialization and readout routines to
+    initialize each variable and capture its accumulated results.
+
+
+    Parameters
+    ----------
+    variables : list of :class:`MonitoredQuantity`
```

Comment:
Could you please make sure that you have quantities everywhere, and not here `variable` and there `quanity`?